src/network/networkd-fdb.c

```diff
@@ -392,3 +399,46 @@ int config_parse_fdb_ntf_flags(
 
         return 0;
 }
+
+int config_parse_fdb_interface(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(fdb_entry_free_or_set_invalidp) FdbEntry *fdb_entry = NULL;
+        Network *network = userdata;
+        int r, ifindex;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = fdb_entry_new_static(network, filename, section_line, &fdb_entry);
+        if (r < 0)
+                return log_oom();
+
+        if (!ifname_valid(rvalue)) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0, "Failed to parse '%s' interface name, ignoring: %s", lvalue, rvalue);
+                return 0;
+        }
+
+        ifindex = resolve_ifname(NULL, rvalue);
+        if (ifindex < 0) {
+                log_error_errno(r, "Could not resolve Interface='%s': %m", rvalue);
+                return 0;
+        }
```

Comment:
this doesn't look right: parsing this at the time the .network files are read might be too late. this should store the ifname as text and resolve when the fdb record is actually installed, so that this works for all network interfaces created between the time networkd reads the .network files and then finally acts on them.

Reply:
Yeah, but we already use resolve_ifname() at many config parsers. We need to fix them at some point...

Reply:
How does it fix ? altname or rename

Reply:
Maybe, store the ifname as is in conf parsers, and then resolve it in configure_xxx().

Reply:
Is it the timing ?

Reply:
Yeah, at that time when we parse the .network files, the corresponding netdevs may not exist yet.

Reply:
But to handle such scenario is not simple. Maybe keep this as is, and leave to a later PR is ok at this time, I think. @poettering WDYT?

Reply:
yes it's strange how to handle this.

Reply:
if this is already a pre-existing pattern it might be Ok to merge this with the current logic, but this really should be fixed I guess. I don#t think the fix is going to be difficult: it just means storing the string the user specified and then at the time of applying things looking for the right ifindex for it. Given that networkd keeps tracks of all ifaces anyway that should be pretty easy.